engine/src/main/java/org/terasology/engine/rendering/primitives/ChunkMesh.java

```diff
@@ -155,8 +156,8 @@ private void generateVBO(RenderType type) {
         VertexElements elements = vertexElements.get(type);
         int id = type.getIndex();
         if (!disposed && elements.finalIndices.limit() > 0 && elements.finalVertices.limit() > 0) {
-            vertexBuffers[id] = bufferPool.get("chunkMesh");
-            idxBuffers[id] = bufferPool.get("chunkMesh");
+            vertexBuffers[id] = bufferPool != null ? bufferPool.get("chunkMesh") : GL15.glGenBuffers();
+            idxBuffers[id] = bufferPool != null ? bufferPool.get("chunkMesh") : GL15.glGenBuffers();
```

Comment:
I don't think we should bother pooling vbo ids. the cost of generating new id's is not that expensive. We end up introducing more traffic to the GPU by populate and depopulate a buffer.  The cost itself occurs when you try write data to a VBO. The commit that introduced this did this to avoid a crash but that is not that case anymore.

Reply:
I'm fine with getting rid of `BufferPool`. Do you think I should do that in this PR? It would simplify some of it a little.

Reply:
depends on how much you have to change. if its easy and simple to do then I think it would make sense here.

Reply:
I think we can do this after in a second PR, might make sense to remove bufferPool in one go.